src/parquet_writer.h

```diff
@@ -0,0 +1,190 @@
+          InstanceMethod("setRowGroupSize", &ParquetWriter::setRowGroupSize)
+        });
+
+    auto constructor = new Napi::FunctionReference();
+    *constructor = Napi::Persistent(func);
+    env.SetInstanceData(constructor);
+
+    exports.Set("ParquetWriter", func);
+    return exports;
+  }
+
+public:
+  ParquetWriter(const Napi::CallbackInfo& info)
+    : Napi::ObjectWrap<ParquetWriter>(info)
+    , builder()
+  {
+    auto env = info.Env();
+    if (info.Length() < 2 || !info[0].IsObject() || !info[1].IsString()) {
+      Napi::TypeError::New(env, "schema::Object, path::string expected").ThrowAsJavaScriptException();
+      return;
+    }
+
+    filepath = info[1].ToString().Utf8Value();
+
+    // Build schema
+    auto schema_obj = info[0].As<Napi::Object>();
+    auto keys = schema_obj.GetPropertyNames();
+    parquet::schema::NodeVector fields;
+    for (uint32_t i = 0; i < keys.Length(); i++) {
+      auto name = keys.Get(i).ToString().Utf8Value();
+      auto type_name = schema_obj.Get(name).ToObject().Get("type").ToString().Utf8Value();
+      auto converted_type = convertedTypeFromString(type_name);
+      auto logical_type = logicalTypeFromString(type_name);
+      columns.push_back(Column{name, logical_type});
+      fields.push_back(parquet::schema::PrimitiveNode::Make(
+        name,
+        parquet::Repetition::OPTIONAL,
+        logical_type,
+        converted_type));
+    }
+    schema = std::static_pointer_cast<parquet::schema::GroupNode>(
+      parquet::schema::GroupNode::Make("schema", parquet::Repetition::OPTIONAL, fields)
+    );
+  }
+
+  Napi::Value appendRow(const Napi::CallbackInfo& info) {
+    auto env = info.Env();
+    if (info.Length() < 1 || !info[0].IsObject()) {
+      Napi::TypeError::New(env, "row::Object expected").ThrowAsJavaScriptException();
+      return Napi::Boolean::New(env, false);
+    }
+
+    auto row = info[0].As<Napi::Object>();
+    for (auto& i : columns) {
+      switch (i.type) {
+      case parquet::Type::INT32:
+        os << row.Get(i.key).ToNumber().Int32Value();
+        break;
+
+      case parquet::Type::INT64:
+        os << row.Get(i.key).ToNumber().Int64Value();
+        break;
+
+      case parquet::Type::FLOAT:
+        os << row.Get(i.key).ToNumber().FloatValue();
+        break;
+
+      case parquet::Type::DOUBLE:
+        os << row.Get(i.key).ToNumber().DoubleValue();
+        break;
+
+      case parquet::Type::BOOLEAN:
+        os << row.Get(i.key).ToBoolean().Value();
+        break;
+
+      case parquet::Type::BYTE_ARRAY:
+        os << row.Get(i.key).ToString().Utf8Value();
+        break;
+
+      default:
+        os << "null";
+      }
+    }
+    os << parquet::EndRow;
+
+    return Napi::Boolean::New(env, true);
+  }
+
+  Napi::Value open(const Napi::CallbackInfo& info) {
+    // Open file
+    std::shared_ptr<arrow::io::FileOutputStream> out_file;
+    PARQUET_ASSIGN_OR_THROW(
+      out_file,
+      arrow::io::FileOutputStream::Open(filepath)
+    );
```

Comment:
This is going to throw a C++ error that might crash JS code, isn't it? This shouldn't happen. You must shield JS from C++.